pkg/config/shortnames/shortnames_config.go

```diff
@@ -0,0 +1,196 @@
+package shortnames
+
+// TODO change this
+
+import (
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	"github.com/pkg/errors"
+)
+
+// PKG TODO:
+// Mutexes
+// More helper functions
+// fullname.search implement
+
+// EXTERNAL TODO:
+// Other conf variables / defaults off << then implement here
+
+// An alias is a short name and it's matching fullname
+// to automatically pull the image from.
+
+// DefaultMode is used when not configured
+const DefaultMode = "search"
+
+type alias struct {
+	Shortname string
+	Fullname  string
+}
+
+// Config is the collection of all aliased short names.
+type Config struct {
+	ShortnameAliasing string `toml:"shortname_aliasing,omitempty"`
+	Alias             []alias
+}
+
+// nameMap is used internally in this package to merge multiple configs.
+type nameMap struct {
+	merged map[string][]string
+}
+
+// modeCondense accepts the alternate names for shortnames.conf shortname_aliasing
+func modeCondense(mode string) int {
+	switch mode {
+	case "0":
+		return 0
+	case "1":
+		return 1
+	case "2":
+		return 2
+	case "3":
+		return 3
+	case "4":
+		return 4
+	case "none":
+		return 0
+	case "false":
+		return 0
+	case "read-only":
+		return 1
+	case "search":
+		return 2
+	case "prompt":
+		return 3
+	case "true":
+		return 3
+	case "always":
+		return 4
+	default:
+		return 2
+	}
+
+}
+
+// squash condenses the information-heavy nameMap into a Config.
+func (nm nameMap) squash(conf *Config) {
+	// Go maps have a non-deterministic order when iterating the keys, so
+	// we dump them in a slice and sort it to enforce some order in
+	// Shortnames slice.  Some consumers of c/image (e.g., CRI-O) log the
+	// the configuration where a non-deterministic order could easily cause
+	// confusion.
+	shortnames := []string{}
+	for shortname := range nm.merged {
+		shortnames = append(shortnames, shortname)
+	}
+	sort.Strings(shortnames)
+
+	// Place into local total version of config
+	conf.Alias = []alias{}
+	for _, shortname := range shortnames {
+		var aliasMerge alias
+		aliasMerge.Shortname = shortname
+		aliasMerge.Fullname = nm.merged[shortname][0]
+		conf.Alias = append(conf.Alias, aliasMerge)
+	}
+}
+
+// prepend creates an alias if it doesn't exist or places it first if it does.
+func (nm *nameMap) prepend(a alias) {
+	var newSlice = []string{a.Fullname}
+	if _, keyExists := nm.merged[a.Shortname]; keyExists {
```

Comment:
```suggestion
	if oldList, keyExists := nm.merged[a.Shortname]; keyExists {
```
and the code below can use `oldList` without repeating the hash lookup so often, for a bit more readable code (and a bit more efficient, but that’s _less_ important).